src/Agent.Listener/Configuration/ServiceControlManager.cs

```diff
@@ -16,6 +16,8 @@ namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
     // TODO: If this pattern repeats, avoid having this conditions and create WindowsServiceLocator/LinuxServiceLocator attribute
     public interface IServiceControlManager : IAgentService
     {
+        void GenerateScripts(AgentSettings settings);
+
```

Comment:
I'm going to refactor this later - the start , stop is never needed - even for windows.  Configure and GenerateScripts can just be factored down to "DoIt()" - unix generates scripts and windows configs it.  Only windows prompts - nix always "does it"